### PR TITLE
feat: convert JSON schema files into device api calls

### PR DESCRIPTION
**Pull request by shakyShane**
**Reviewer:** @GioSensation 
**Asana:** https://app.asana.com/0/0/1202360827060035/f

## Description

I extracted this to be reviewed in isolation, without the noise of the incoming schemas 💪🏻  Please read the documentation + tests provided in the PR.

- [docs](https://github.com/duckduckgo/duckduckgo-autofill/blob/shane/schema-generated-files/docs/device-api-calls.md)
- [unit tests](https://github.com/duckduckgo/duckduckgo-autofill/blob/500e089644e01003feb93f440c621c6efac5622e/scripts/tests/api-call-generator.test.js)

### Please note: 

3,402 lines of the 'changed' lines are from the package-lock.json file 🙈

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,8 @@
     "zod": "^3.16.0"
   },
   "dependencies": {
-    "@duckduckgo/content-scope-scripts": "github:duckduckgo/content-scope-scripts#1.3.0"
+    "@duckduckgo/content-scope-scripts": "github:duckduckgo/content-scope-scripts#1.3.0",
+    "json-schema-to-typescript": "^10.1.5",
+    "ts-to-zod": "^1.10.0"
```
Comment (GioSensation) [inline package.json:67]: Just double checking that these need to be in the dependencies vs the dev-dependencies.

File: scripts/api-call-generator.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    if (cd.resultValidator && cd.resultValidator !== 'any') {
+        lines.push(`  resultValidator = ${cd.resultValidator}`)
+    }
+    lines.push(`}`)
+    return lines.join('\n')
+}
+
+// the validator for the incoming JSON files
+
+/**
```
Comment (GioSensation) [inline scripts/api-call-generator.js:232]:
```suggestion
/**
 * The validator for the incoming JSON files
```

**Review by GioSensation — APPROVED** [review]
Love the documentation. Especially, the summaries with examples. Just a couple of minor things, but this is approved. Thank you. 🚀

File: package.json (hunk 1 above)
Comment (shakyShane) [reply package.json]: Fixed, thanks for catching that!

**Pull request merged**